### PR TITLE
Facebook login

### PR DESCRIPTION
**Pull request by dani5447**

### CODE REVIEW

File: app/js/index.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 import react from 'react';
 
-console.log(react);
+console.log(react);
```
Comment (JRJurman) [inline app/js/index.js]: should maybe remove?

File: package.json [hunk 2]
```diff
@@ -19,6 +19,8 @@
     "gulp-util": "^3.0.6",
     "mocha": "^2.3.2",
     "react": "^0.13.3",
-    "vinyl-source-stream": "^1.1.0"
+    "vinyl-source-stream": "^1.1.0",
+    "passport": ">= 0.0.0",
```
Comment (bbesmanoff) [inline package.json]: May want to provide a more restrictive version

File: server.js [hunk 3]
```diff
@@ -1,9 +1,120 @@
...
+// Initialize Passport!  Also use passport.session() middleware, to support
+// persistent login sessions (recommended).
+server.use(passport.initialize());
+server.use(passport.session());
+
+
+
+//TODO may want these secure somewhere?
+var FACEBOOK_APP_ID = "819934004772176";
+var FACEBOOK_APP_SECRET = "90ff6c2c0231f2b185acb5ecdc74dfae";
```
Comment (bbesmanoff) [inline server.js:23]: We can set up [dotenv](https://www.npmjs.com/package/dotenv)

File: server.js [hunk 4]
```diff
@@ -1,9 +1,120 @@
...
+});
+
+// Use the FacebookStrategy within Passport.
+//   Strategies in Passport require a `verify` function, which accept
+//   credentials (in this case, an accessToken, refreshToken, and Facebook
+//   profile), and invoke a callback with a user object.
+passport.use(new FacebookStrategy({
+    clientID: FACEBOOK_APP_ID,
+    clientSecret: FACEBOOK_APP_SECRET,
+    callbackURL: "http://localhost:3000/auth/facebook/callback"
```
Comment (bbesmanoff) [inline server.js:50]: Will this change when it's hosted on a server?  I can see it being fine, it just feels weird to read `localhost`

File: server.js [hunk 5]
```diff
@@ -1,9 +1,120 @@
...
+             function(req, res) { 
+      console.log('authentication passed, redirecting to home page');
+      res.redirect('/home');
+  });    
+    
+  server.get('/logout', function(req, res){
+        req.logout();
+        res.redirect('/login');
+  });
+
```
Comment (bbesmanoff) [inline server.js]: Probably want to move these out of (`if (app.get('development'))`)

File: server.js [hunk 6]
```diff
@@ -1,9 +1,120 @@
...
 if (server.get('env') !== 'production') {
   server.use('/', express.static('dist'));
+    
+  server.get('/login', ensureAuthenticated, function (req, res) {
+    res.redirect('/home');
+  });
+    
+  server.get('/home', function (req, res) {
+    res.sendFile('home.html',  {root: path.join(__dirname, 'dist')})
+  });
```
Comment (bbesmanoff) [inline server.js]: This may change with #11 and when components start coming in

File: server.js (hunk 3 above)
Comment (bbesmanoff) [inline server.js:23]: Will be pulling this into a separate issue

**Pull request merged**